jack-and-the-beans/buddies-ios
# BUD-14: View activities

This PR introduces these features:
- Tap on an activity list entry and it will modally display an activity
- See all of the information about an activity (except location, TBD because of Grant's new work)
- If you are NOT a member or owner of the activity, it shows you the public view with join button (~not working~ NOW WORKING)
- If you are a member or owner of the activity, it displays a dummy chat screen (WIP)
- Update: JOIN ACTIVITIES !!!!

This PR introduces these internal functions:
- `useUsers`, which allows you to get updates for an array of users instead of a single user
- Add UIImage to the user model and populate it up-front. That way it's always available.
- Add an extension on UIView which fits any view to the bounds of its superview (e.g. fills the entire space of its parent view)
- Adds an extension to UIImageView (`makeCircle`) which makes the image circular
- Adds a getter to Activity which returns the `TimeInterval` for the activity.

**This PR has no new tests.**
_Also, apologies for the messy branch history. There were a lot of things going on at the same time._

To test:
- Checkout the branch and run it
- Navigate to any topic and tap on a few different kinds of activities:
    - Ones which you own (should show the temp screen)
    - Ones which you aren't a member of (should show the public screen)
    - Ones with varying information like multiple topics, or only one topic
- Specifically, the activity `Bible Study > Cool Beans Buddies` should do a few things:
    - You should see the public view since I own it
    - You should see how the topics collection works with a large number of topics
    - You should see a lengthy description which gets cut off (perhaps we could improve this)
    - Tap the join button. You should immediately see the "joined" screen
    - Once you join, and if you navigate back to the list view, you should see your pic on the list entry for the activity
